smsapi/Api/Action/Sender/List.cs

```diff
@@ -1,23 +1,29 @@
-﻿using System.Collections.Specialized;
+﻿using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.IO;
 using SMSApi.Api.Response;
 
 namespace SMSApi.Api.Action
 {
-    public class SenderList : BaseArray<Sender>
+    public class SenderList : Base<Array<Sender>>
     {
+        protected override Array<Sender> ResponseToObject(Stream data)
+        {
+            return new Array<Sender>(Deserialize<List<Sender>>(data));
+        }
+
         protected override string Uri()
         {
             return "sender.do";
         }
 
         protected override NameValueCollection Values()
         {
-            var collection = new NameValueCollection();
-
-            collection.Add("format", "json");
-            collection.Add("list", "1");
-
-            return collection;
+            return new NameValueCollection
+            {
+                { "format", "json" },
```

Comment:
You can extract this to method in base class, which gives you collection with json to avoid adding json in each action class.

Reply:
35c5ffe